workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -129,5 +130,214 @@ private static bool TryGetApkPath(string rootPath, out string apkPath)
+
+                    if (RedirectedProcess.Command("xcrun")
+                        .WithArgs("instruments", "-w", deviceUID, "-t", "Blank")
+                        .Run() != 0)
+                    {
+                        Debug.LogError("Was unable to start iOS Simulator.");
+                        return;
+                    }
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing your app", 0.7f);
+                var command = "osascript";
+                var commandArgs = $@"-e 'tell application ""Terminal""
+                                     activate
+                                     do script ""xcodebuild test-without-building -destination id={deviceUID} -xctestrun {xcTestRunPath}""
+                                     end tell'";
+
+                var processInfo = new ProcessStartInfo(command, commandArgs)
+                {
+                    CreateNoWindow = false,
+                    UseShellExecute = true,
+                    WorkingDirectory = Common.SpatialProjectRootDir
+                };
+
+                var process = Process.Start(processInfo);
+
+                if (process == null)
+                {
+                    Debug.LogError("Failed to start app on iOS Simulator.");
+                    return;
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
+            }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+
+        private static bool TryGetXCTestRunPath(bool forSimulator, out string xctestrunPath)
+        {
+            foreach (var file in Directory.GetFiles(DerivedDataPath, "*.xctestrun", SearchOption.AllDirectories))
+            {
+                if (!forSimulator && file.Contains("iphoneos"))
+                {
+                    xctestrunPath = file;
+                    return true;
+                }
+
+                if (forSimulator && file.Contains("iphonesimulator"))
+                {
+                    xctestrunPath = file;
+                    return true;
+                }
+            }
+
+            xctestrunPath = string.Empty;
+            return false;
+        }
+
+        private static bool TryPrepareArguments(bool shouldConnectLocally, out string builtArguments)
+        {
+            var gdkToolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
+            var arguments = new StringBuilder();
+            builtArguments = string.Empty;
+            if (shouldConnectLocally)
+            {
+                if (string.IsNullOrEmpty(gdkToolsConfig.RuntimeIp))
+                {
+                    Debug.LogWarning("No local runtime IP was specified. Ensure you set one in SpatialOS > GDK tools configuration.");
+                }
+
+                arguments.Append($"+{RuntimeConfigNames.Environment} {RuntimeConfigDefaults.LocalEnvironment} ");
```

Comment:
I'm not a fan of adding the trailing spaces at the end of each `Append` string. Makes it really error prone. 

Could have a local const `const string argStructure = "+{0} {1} ";` then call `argStructure.Format(argName, argValue);`?